core/web/keys_controller.go

```diff
@@ -67,6 +71,21 @@ func (kc *keysController[K, R]) Create(c *gin.Context) {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 		return
 	}
+
+	// Emit audit log, determine if Terra or Solana key
+	switch unwrappedKey := any(key).(type) {
+	case terrakey.Key:
+		kc.lggr.Audit(audit.TerraKeyCreated, map[string]interface{}{
+			"publicKey": unwrappedKey.PublicKey(),
+			"id":        unwrappedKey.ID(),
+		})
+	case solkey.Key:
+		kc.lggr.Audit(audit.SolanaKeyCreated, map[string]interface{}{
+			"publicKey": unwrappedKey.PublicKey(),
+			"id":        unwrappedKey.ID(),
+		})
```

Comment:
And what if it is neither?

Reply:
Right, we have a discussion above for this: https://github.com/smartcontractkit/chainlink/pull/6782#discussion_r894474630
but for the way it is now, a new chain would require creating new audit log event types for and adding them to any new handlers created, either explicitly or here through this abstraction. I was looking to conflate the audit log events into a single string constant, but we still need to access the .PublicKey() portion of the key, which isn't part of the Key interface because of the different implementations. Looking at the tradeoff this way for now